search_queries_interval.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2012-present Oliver Eilhard. All rights reserved.
+// Use of this source code is governed by a MIT-license.
+// See http://olivere.mit-license.org/license.txt for details.
+
+package elastic
+
+// IntervalRule represents the generic matching rule interface. Matching Rule is
+// just a Query, but may be used only inside IntervalQuery. An extra method is added
+// just to separate *Rule objects from other queries
+type IntervalRule interface {
+	Query
+
+	// IsIntervalRule is never actually called, and is used just for Rule to differ from standard Query
+	IsIntervalRule() bool
+}
+
+// IntervalQuery returns documents based on the order and proximity of matching terms
+//
+// For more details, see
+// https://www.elastic.co/guide/en/elasticsearch/reference/7.4/query-dsl-intervals-query.html
+type IntervalQuery struct {
+	field string
+	rule  IntervalRule
+}
+
+// NewIntervalQuery creates and initializes a new IntervalQuery.
+func NewIntervalQuery(field string, rule IntervalRule) *IntervalQuery {
```

Comment:
There can be only one `IntervalQueryRule`, correct?

Reply:
Do you mean if `rule ..IntervalRule` should be possible?
I guess no, but I'll check in java sources too.

The idea is that you add just one rule object. If you need complex logic: you use `all_of`, `any_of`, etc then can add OR/ADD logic to your multiple rules

Reply:
Yes, that's what I meant. Maybe you can do `rule ...IntervalQueryRule`... I haven't looked into the Java source yet.

Reply:
in Java sources it is called `IntervalsSourceProvider` and it can be just one. So nothing is changed here.